treeherder/perf/outcome_checker.py

```diff
@@ -0,0 +1,68 @@
+from django.db.models import Q, QuerySet
+
+from enum import Enum
+from typing import List
+
+from treeherder.model.models import Job, Push
+from treeherder.perf.models import (
+    BackfillRecord,
+)
+
+class OutcomeStatus(Enum):
+    SUCCESSFUL = 1
+    FAILED = 2
+    IN_PROGRESS = 3
+
+#TODO delete this after soft launch lands (job_type will come from BackfillRecord)
+def _get_job_type(record):
+    record_context = record.get_context()
+    job_type = None
+
+    if len(record_context) >= 1:
+        job_id = record_context[0]["job_id"]
+        job = Job.objects.get(id=job_id)
+        if job:
+            job_type = job.job_type
+
+    return job_type
+
+
+class OutcomeChecker:
+    """
+    Checks outcome of backfills
+    """
+
+    def __init__(self):
+        pass
+
+    def check(self, record: BackfillRecord) -> int:
+        # TODO: get job_type from record when soft launch lands ---> job_type = record.job_type
+        job_type = _get_job_type(record)
+        from_time, to_time = self._get_push_timestamp_range(record.get_context())
+        repository_id = record.alert.summary.repository.id
+        pushes_in_range = self._get_pushes_in_range(from_time, to_time, repository_id)
```

Comment:
What do you think of moving this method to `BackfillRecord`? There're definitely 2 client codes that could benefit from this.

Reply:
I support this idea, I'll create a bug for extracting this